### DISCUSSION TITLE
What is the major differences between files.

### DISCUSSION THREAD

**Discussion by wadakatu** [Laravel-Lang/common#8] [Q&A] [answered]
Hello.
I would like to know the major differences between `json.json`, `json-inline.json`, `php.json` and `php-inline.json`.                    

In my opinion...
```
php.json and php-inline.json are for the translation of validation.

json.json and json-inline are for the translation of words that are used in laravel, laravel nova etc...
```

Please let me know if I am wrong.
Thank you.

**Comment by andrey-helldar** [comment, marked as answer]
Hi!

**php.json** - contains translations of values for PHP localization files.
**php-inline.json** - contains inline translations of values for PHP localization files.

**json.json** - contains translations of values for JSON localization files.
**json-inline.json** - contains inline translations of values for JSON localization files.

At the same time, the Laravel Framework itself also contains JSON keys:

- https://github.com/laravel/framework/blob/9.x/src/Illuminate/Auth/Notifications/VerifyEmail.php#L65-L68
- https://github.com/Laravel-Lang/lang/blob/main/source/framework/9.x/framework.json

At the same time, the project structure is not tied to specific packages or files. For example, the [Laravel-Lang/http-statuses](https://github.com/Laravel-Lang/http-statuses) project contains translations of statuses, while [Laravel-Lang/attributes](https://github.com/Laravel-Lang/attributes) contains attributes for validation files.

Using the [template](https://github.com/Laravel-Lang/translations-template), you can create any project with any translation keys.

This file structure allows you to support translation for several product versions at once. For example, [Jetstream](https://github.com/Laravel-Lang/lang/tree/main/source/jetstream) `1.x`. `2.x` and `dev` versions.

**Reply by wadakatu** [reply]
@andrey-helldar 

Thank you for your answer.

I understood what these files are.
However, I am not quite sure what inline translations are.
What is inline translations?
Can you give me some example about inline translation and non-inline translation?

**Reply by andrey-helldar** [reply]
Main:

![image](https://user-images.githubusercontent.com/10347617/207378059-c1c2e0cd-0c45-48b7-8096-3fc8ccb63468.png)

Inline:

![image](https://user-images.githubusercontent.com/10347617/207378130-7aa2d78b-485d-40c7-8c64-c0e059f599ad.png)
